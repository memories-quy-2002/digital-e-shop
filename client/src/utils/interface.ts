export enum Role {
	Customer = "Customer",
	Admin = "Admin",
}

export interface User {
	UID: number;
	UName: string;
	UEmail: string;
	UPassword: string;
	URole: Role;
	ULast_login: string;
}
