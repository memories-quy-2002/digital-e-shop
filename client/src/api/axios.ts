import axios from "axios";
declare module "axios" {
    export interface AxiosRequestConfig {
        handlerEnabled?: boolean;
    }
}

const baseURL =
    process.env.NODE_ENV === "production"
        ? "https://e-commerce-express-server-app.vercel.app/"
        : "http://localhost:4000";

export default axios.create({
    baseURL,
    withCredentials: true,
    handlerEnabled: true,
});
