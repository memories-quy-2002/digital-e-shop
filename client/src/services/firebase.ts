// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getAuth } from "firebase/auth";
// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
    apiKey: "AIzaSyCae88IRpKYJbHLxZIiArzIPYTkglQqgb0",
    authDomain: "graduation-project-5bbfb.firebaseapp.com",
    projectId: "graduation-project-5bbfb",
    storageBucket: "graduation-project-5bbfb.appspot.com",
    messagingSenderId: "503526214575",
    appId: "1:503526214575:web:5c4e1263f106bc2bee7d5a",
    measurementId: "G-NGN3CY83D3",
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
export const auth = getAuth(app);
