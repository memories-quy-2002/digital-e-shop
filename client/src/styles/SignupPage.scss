.signup {
	width: 90%;
	background-color: white;
	display: flex;
	flex-direction: row;
	height: 900px;
	box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1), 0px 4px 8px rgba(0, 0, 0, 0.05);
	&:hover {
		box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.2);
	}

	&__image {
		flex: 2;
		position: relative;
		&__background {
			position: absolute;
			height: 100%;
			width: 100%;
			background: url("../assets/images/background_form.jpg");
			object-fit: cover;
			mix-blend-mode: normal;
			filter: brightness(0.4);
		}
		&__content {
			width: 100%;
			height: 100%;
			&__name {
				width: inherit;
				position: absolute;
				left: 0%;
				right: 56.98%;
				top: 36.39%;
				bottom: 45.46%;
				font-weight: 600;
				font-size: 60px;
				line-height: 116px;
				text-align: center;
				color: #ffffff;
				text-shadow: 0px 12px 4px rgba(0, 0, 0, 0.25);
			}
			&__desc {
				width: inherit;
				position: absolute;
				left: 0%;
				right: 56.98%;
				top: 52.13%;
				bottom: 36.48%;
				font-weight: 500;
				font-size: 36px;
				line-height: 44px;
				text-align: center;
				color: #ffffff;

				text-shadow: 0px 12px 4px rgba(0, 0, 0, 0.25);
			}
		}
	}

	&__form {
		height: fit-content;
		flex: 3;
		display: flex;
		flex-direction: column;
		align-items: center;
		&__title {
			font-size: 36px;
			font-weight: bold;
			color: black;
			margin-top: 2rem;
			margin-bottom: 3rem;
		}
		&__container {
			width: 80%;
			&__group {
				width: 100%;
				&__input {
					padding: 0.5rem;
					box-shadow: inset 0px 1px 1px 1px rgba(0, 0, 0, 0.25);
				}
			}
			input[type="radio"] {
				border: 1px solid rgba(0, 0, 0, 0.25);
			}
		}
	}
}
