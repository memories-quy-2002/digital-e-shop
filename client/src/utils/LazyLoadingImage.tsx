import { useEffect, useRef } from "react";

const LazyLoadImage = ({ src, alt }: { src: string; alt: string }) => {
    const imgRef = useRef<HTMLImageElement | null>(null);

    useEffect(() => {
        const observer = new IntersectionObserver(
            (entries) => {
                entries.forEach((entry) => {
                    if (entry.isIntersecting && imgRef.current) {
                        imgRef.current.src = imgRef.current.dataset.src!;
                        observer.unobserve(imgRef.current);
                    }
                });
            },
            { threshold: 0.05 } // Load when 10% of the image is visible
        );

        if (imgRef.current) observer.observe(imgRef.current);
    }, []);

    return <img ref={imgRef} data-src={src} alt={alt} loading="lazy" />;
};
export default LazyLoadImage;
