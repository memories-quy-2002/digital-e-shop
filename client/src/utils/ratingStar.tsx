import { BsStar, BsStarFill, BsStarHalf } from "react-icons/bs";

const ratingStar = (overallScore: number, color: string = "#FFCC4A") => {
	const stars = [];
	const maxStars = 5;

	for (let i = 1; i <= maxStars; i++) {
		if (i <= overallScore) {
			stars.push(
				<div key={i} className="d-flex align-items-center">
					<BsStarFill color={color} />
				</div>
			); // Full star
		} else if (i - 1 < overallScore && overallScore < i) {
			stars.push(
				<div key={i} className="d-flex align-items-center">
					<BsStarHalf color={color} />
				</div>
			); // Half star
		} else {
			stars.push(
				<div key={i} className="d-flex align-items-center">
					<BsStar color={color} />{" "}
				</div>
			); // Empty star
		}
	}

	return stars;
};
export default ratingStar;
