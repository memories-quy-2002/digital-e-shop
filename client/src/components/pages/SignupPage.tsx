import { createUserWithEmailAndPassword } from "firebase/auth";
import React, { useState } from "react";
import { Button, Container, Form } from "react-bootstrap";
import { Link, useNavigate } from "react-router-dom";
import axios from "../../api/axios";
import { auth } from "../../services/firebase";
import "../../styles/SignupPage.scss";
import { Role } from "../../utils/interface";
import Cookies from "universal-cookie";
import { Helmet } from "react-helmet";

const cookies = new Cookies();

console.log(cookies.get("rememberMe"));

interface User {
    username: string;
    email: string;
    password: string;
    confirm: string;
    role: Role;
}
const SignupPage = () => {
    const navigate = useNavigate();
    const [user, setUser] = useState<User>({
        username: "",
        email: "",
        password: "",
        confirm: "",
        role: Role.Customer,
    });

    const [errors, setErrors] = useState<string[]>([]);

    const validateForm = (): string[] => {
        const errorsList: string[] = [];
        const usernamePattern = /^[a-zA-Z0-9._-]{3,15}$/; // Allow letters, numbers, ., _, -, length 3-15
        const passwordPattern = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
        const emailPattern = /^([A-Za-z0-9_\-.])+@([A-Za-z0-9_\-.])+\.([A-Za-z]{2,4})$/;
        if (!user.username) {
            errorsList.push("Username is required");
        } else if (!usernamePattern.test(user.username)) {
            errorsList.push(
                "Username must be 3-15 characters long and contain only letters, numbers, periods, underscores, or hyphens."
            );
        }
        if (!user.email) {
            errorsList.push("Email is required");
        } else if (!user.email.match(emailPattern)) {
            errorsList.push("Invalid email format");
        }
        if (!user.password) {
            errorsList.push("Password is required");
        } else if (!passwordPattern.test(user.password)) {
            errorsList.push(
                "Password must be at least 8 characters long, contain at least one lowercase letter, one uppercase letter, one number, and one special character."
            );
        } else if (!user.confirm) {
            errorsList.push("Confirm password is required");
        } else if (user.password !== user.confirm) {
            errorsList.push("Confirm password is not match");
        }
        return errorsList;
    };

    const handleChangeInput = (event: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = event.target;
        setUser({ ...user, [name]: value });
    };
    const handleChangeRadio = (event: React.ChangeEvent<HTMLInputElement>) => {
        const selectedRole = event.target.value as Role;
        setUser({ ...user, role: selectedRole });
    };
    const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        const errorsList: string[] = validateForm();

        if (errorsList.length > 0) {
            setErrors(errorsList); // Display errors, no need to proceed further
            return;
        } else {
            try {
                const userCredential = await createUserWithEmailAndPassword(auth, user.email, user.password);
                const uid = userCredential.user.uid;
                const response = await axios.post("/api/users", {
                    user,
                    uid,
                });
                if (response.status === 200) {
                    const token = response.data.token;
                    const cookieData = {
                        uid,
                        token,
                    };
                    cookies.set("rememberMe", JSON.stringify(cookieData), {
                        httpOnly: false,
                        // Consider using Secure flag if using HTTPS
                        maxAge: 1000 * 60 * 60 * 24 * 30,
                    });
                    if (user.role === Role.Customer) {
                        navigate("/");
                    } else if (user.role === Role.Admin) {
                        navigate("/admin");
                    }
                }
            } catch (err: any) {
                if (err.response) {
                    const status = err.response.status;
                    setErrors([err.response.data.msg]);
                    if (status === 401) {
                        // Handle 401 Unauthorized
                        console.error("Unauthorized access. Please check your credentials.");
                    } else if (status === 500) {
                        // Handle 500 Internal Server Error
                        console.error("Internal Server Error. Please try again later.");
                    } else {
                        // Handle other errors
                        console.error(`Error: ${err.response.status}`);
                    }
                } else {
                    // Handle non-Axios errors
                    console.error(err.message);
                    setErrors(["An unexpected error occurred."]);
                }
            }
        }
    };

    return (
        <Container
            fluid
            style={{
                backgroundColor: "#E6F0FD",
                padding: "3rem",
                display: "flex",
                justifyContent: "center",
            }}
        >
            <Helmet>
                <title>Signup</title>
            </Helmet>
            <div className="signup">
                <div className="signup__image">
                    <img
                        src={require("../../assets/images/background_form.jpg")}
                        alt=""
                        className="signup__image__background"
                    ></img>
                    <div className="signup__image__content">
                        <strong className="signup__image__content__name">DIGITAL-E</strong>
                        <p className="signup__image__content__desc">An E-commerce platforms of electronics devices</p>
                    </div>
                </div>
                <div className="signup__form">
                    <h3 className="signup__form__title">Create new account</h3>
                    <Form
                        className="signup__form__container"
                        onSubmit={handleSubmit}
                        name="signup-form"
                        aria-label="signup-form"
                    >
                        <Form.Group className="signup__form__container__group mb-3" controlId="formBasicUserName">
                            <Form.Label>Username</Form.Label>
                            <Form.Control
                                type="text"
                                name="username"
                                placeholder="Username"
                                className="signup__form__container__group__input"
                                required
                                autoComplete="off"
                                value={user.username}
                                onChange={handleChangeInput}
                            />
                        </Form.Group>
                        <Form.Group className="signup__form__container__group mb-3" controlId="formBasicEmail">
                            <Form.Label>Email address</Form.Label>
                            <Form.Control
                                type="email"
                                name="email"
                                placeholder="Email"
                                className="signup__form__container__group__input"
                                required
                                autoComplete="off"
                                value={user.email}
                                onChange={handleChangeInput}
                            />
                        </Form.Group>
                        <Form.Group className="signup__form__container__group mb-3" controlId="formBasicPassword">
                            <Form.Label>Password</Form.Label>
                            <Form.Control
                                type="password"
                                name="password"
                                placeholder="Password"
                                className="signup__form__container__group__input"
                                required
                                autoComplete="off"
                                value={user.password}
                                onChange={handleChangeInput}
                            />
                        </Form.Group>
                        <Form.Group
                            className="signup__form__container__group mb-3"
                            controlId="formBasicConfirmPassword"
                        >
                            <Form.Label>Confirm Password</Form.Label>
                            <Form.Control
                                type="password"
                                name="confirm"
                                placeholder="Confirm Password"
                                className="signup__form__container__group__input"
                                required
                                autoComplete="off"
                                value={user.confirm}
                                onChange={handleChangeInput}
                            />
                        </Form.Group>
                        <Form.Group className="signup__form__container__group mb-3 mt-5">
                            <Form.Label>Signup as:</Form.Label>
                            <br />
                            <Form.Check
                                inline
                                type="radio"
                                name="signup-customer"
                                id="customer-radio"
                                value={Role.Customer}
                                checked={user.role === Role.Customer}
                                onChange={handleChangeRadio}
                            />
                            <Form.Label htmlFor="customer-radio" className="me-3">
                                Customer
                            </Form.Label>

                            <Form.Check
                                inline
                                type="radio"
                                name="signup-admin"
                                id="admin-radio"
                                value={Role.Admin}
                                checked={user.role === Role.Admin}
                                onChange={handleChangeRadio}
                            />
                            <Form.Label htmlFor="admin-radio">Admin</Form.Label>
                        </Form.Group>

                        <div className="mb-5">
                            {errors.map((error, id) => (
                                <div key={id} className="text-danger mb-2">
                                    {error}
                                </div>
                            ))}
                        </div>
                        <Button
                            variant="primary"
                            type="submit"
                            style={{
                                width: "100%",
                                height: "3rem",
                                fontSize: "16px",
                                textTransform: "uppercase",
                            }}
                        >
                            Sign up
                        </Button>
                        <div className="mt-4" style={{ textAlign: "center" }}>
                            Already registered? <Link to="/login">Login</Link>
                        </div>
                    </Form>
                </div>
            </div>
        </Container>
    );
};

export default SignupPage;
