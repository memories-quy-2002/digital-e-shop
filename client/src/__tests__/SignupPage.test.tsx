import { vi, describe, it, expect, beforeEach, afterEach, Mock } from "vitest";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom/vitest";
import { createUserWithEmailAndPassword } from "firebase/auth";
import { MemoryRouter } from "react-router-dom";
import axios from "../api/axios";
import SignupPage from "../components/pages/SignupPage";
import { auth } from "../services/firebase";
import { Role } from "../utils/interface";
vi.mock("firebase/auth", () => ({
    getAuth: vi.fn(),
    createUserWithEmailAndPassword: vi.fn(),
}));
vi.mock("../api/axios");

const mockedAxios = vi.mocked(axios);
declare global {
    interface Performance {
        markResourceTiming: Mock;
    }
}

const handleOnSubmitMock = vi.fn();
describe("SignupPage", () => {
    const users = {
        customer: {
            username: "customer123",
            email: "test6@gmail.com",
            password: "Memories2002@",
            confirm: "Memories2002@",
            role: Role.Customer,
        },
        admin: {
            username: "admin123",
            email: "test7@gmail.com",
            password: "Memories2002@",
            confirm: "Memories2002@",
            role: Role.Admin,
        },
    };

    beforeEach(() => {
        vi.clearAllMocks();
    });

    const renderSignupPage = (user: typeof users.customer) => {
        render(
            <MemoryRouter>
                <SignupPage />
            </MemoryRouter>
        );

        // Fill in email and password

        screen.getByRole("form", { name: "signup-form" }).onsubmit = handleOnSubmitMock;
        fireEvent.change(screen.getByPlaceholderText("Username"), {
            target: { value: user.username },
        });
        fireEvent.change(screen.getByPlaceholderText("Email"), {
            target: { value: user.email },
        });
        fireEvent.change(screen.getByPlaceholderText("Password"), {
            target: { value: user.password },
        });

        fireEvent.change(screen.getByPlaceholderText("Confirm Password"), {
            target: { value: user.confirm },
        });

        // Select role (Customer or Admin)
        if (user.role === Role.Customer) {
            const customerRadio = screen.getByLabelText("Customer");
            fireEvent.click(customerRadio);
            expect(customerRadio).toBeChecked();
        } else if (user.role === Role.Admin) {
            const adminRadio = screen.getByLabelText("Admin");
            fireEvent.click(adminRadio);
            expect(adminRadio).toBeChecked();
        }
        fireEvent.click(screen.getByRole("button", { name: "Sign up" }));
    };

    it("should match the SignupPage snapshot", async () => {
        const { asFragment } = render(
            <MemoryRouter>
                <SignupPage />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("should handle successful customer signup", async () => {
        (createUserWithEmailAndPassword as Mock).mockResolvedValue({
            user: { uid: "mock-user-id" },
        });
        // Mock axios response
        (mockedAxios.post as Mock).mockImplementationOnce(() =>
            Promise.resolve({
                status: 200,
                data: { token: "customerMockedToken" },
            })
        );

        renderSignupPage(users.customer);
        expect(screen.getByText("Create new account")).toBeInTheDocument();
        expect(handleOnSubmitMock).toHaveBeenCalled();
        await waitFor(() => {
            expect(createUserWithEmailAndPassword).toHaveBeenCalledWith(
                auth,
                users.customer.email,
                users.customer.password
            );
        });

        // Wait for the axios post call
        await waitFor(() => {
            expect(mockedAxios.post).toHaveBeenCalledWith("/api/users", {
                user: users.customer,
                uid: "mock-user-id",
            });
        });
    });

    it("should handle successful admin signup", async () => {
        // Mock axios response
        (createUserWithEmailAndPassword as Mock).mockResolvedValue({
            user: { uid: "mock-user-id" },
        });
        (mockedAxios.post as Mock).mockImplementationOnce(() =>
            Promise.resolve({
                status: 200,
                data: { token: "adminMockedToken" },
            })
        );

        renderSignupPage(users.admin);

        expect(screen.getByText("Create new account")).toBeInTheDocument();
        expect(handleOnSubmitMock).toHaveBeenCalled();
        await waitFor(() => {
            expect(createUserWithEmailAndPassword).toHaveBeenCalledWith(auth, users.admin.email, users.admin.password);
        });
        // Wait for the axios post call
        await waitFor(() => {
            expect(mockedAxios.post).toHaveBeenCalledWith("/api/users", {
                user: users.admin,
                uid: "mock-user-id",
            });
        });
    });

    it("should handle signup error", async () => {
        // Mock axios to return an error
        (mockedAxios.post as Mock).mockImplementationOnce(() =>
            Promise.reject({
                status: 401,
                data: { token: "customerMockedToken" },
            })
        );
        renderSignupPage({
            username: "customer123",
            email: "test@example.com",
            password: "wrong_password",
            confirm: "wrong_password",
            role: Role.Customer,
        });
        expect(screen.getByText("Create new account")).toBeInTheDocument();
        // Check if the error message is shown
        await waitFor(() => {
            expect(
                screen.getByText(
                    /Password must be at least 8 characters long, contain at least one lowercase letter, one uppercase letter, one number, and one special character./i
                )
            ).toBeInTheDocument();
        });
    });

    it("should handle empty email", async () => {
        // Mock axios to return an error
        (mockedAxios.post as Mock).mockImplementationOnce(() =>
            Promise.reject({
                status: 401,
                data: { token: "customerMockedToken" },
            })
        );
        renderSignupPage({
            username: "customer123",
            email: "",
            password: "wrong_password",
            confirm: "wrong_password",
            role: Role.Customer,
        });
        expect(screen.getByText("Create new account")).toBeInTheDocument();
        // Check if the error message is shown
        await waitFor(() => {
            expect(screen.getByText(/Email is required/i)).toBeInTheDocument();
        });
    });

    it("should handle invalid email", async () => {
        // Mock axios to return an error
        (mockedAxios.post as Mock).mockImplementationOnce(() =>
            Promise.reject({
                status: 401,
                data: { token: "customerMockedToken" },
            })
        );
        renderSignupPage({
            username: "customer123",
            email: "test1@",
            password: "wrong_password",
            confirm: "wrong_password",
            role: Role.Customer,
        });
        expect(screen.getByText("Create new account")).toBeInTheDocument();
        // Check if the error message is shown
        await waitFor(() => {
            expect(screen.getByText(/Invalid email format/i)).toBeInTheDocument();
        });
    });

    it("should handle empty password", async () => {
        // Mock axios to return an error
        (mockedAxios.post as Mock).mockImplementationOnce(() =>
            Promise.reject({
                status: 401,
                data: { token: "adminMockedToken" },
            })
        );

        renderSignupPage({
            username: "admin123",
            email: "test@gmail.com",
            password: "",
            confirm: "",
            role: Role.Admin,
        });
        expect(screen.getByText("Create new account")).toBeInTheDocument();
        // Check if the error message is shown
        await waitFor(() => {
            expect(screen.getByText(/Password is required/i)).toBeInTheDocument();
        });
    });
});
