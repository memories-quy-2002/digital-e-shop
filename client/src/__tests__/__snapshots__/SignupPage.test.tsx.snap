// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupPage matches the SignupPage snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-fluid"
    style="background-color: rgb(230, 240, 253); padding: 3rem; display: flex; justify-content: center;"
  >
    <div
      class="signup"
    >
      <div
        class="signup__image"
      >
        <img
          alt=""
          class="signup__image__background"
          src="background_form.jpg"
        />
        <div
          class="signup__image__content"
        >
          <strong
            class="signup__image__content__name"
          >
            DIGITAL-E
          </strong>
          <p
            class="signup__image__content__desc"
          >
            An E-commerce platforms of electronics devices
          </p>
        </div>
      </div>
      <div
        class="signup__form"
      >
        <h3
          class="signup__form__title"
        >
          Create new account
        </h3>
        <form
          aria-label="signup-form"
          class="signup__form__container"
          name="signup-form"
        >
          <div
            class="signup__form__container__group mb-3"
          >
            <label
              class="form-label"
              for="formBasicUserName"
            >
              Username
            </label>
            <input
              autocomplete="off"
              class="signup__form__container__group__input form-control"
              id="formBasicUserName"
              name="username"
              placeholder="Username"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="signup__form__container__group mb-3"
          >
            <label
              class="form-label"
              for="formBasicEmail"
            >
              Email address
            </label>
            <input
              autocomplete="off"
              class="signup__form__container__group__input form-control"
              id="formBasicEmail"
              name="email"
              placeholder="Email"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="signup__form__container__group mb-3"
          >
            <label
              class="form-label"
              for="formBasicPassword"
            >
              Password
            </label>
            <input
              autocomplete="off"
              class="signup__form__container__group__input form-control"
              id="formBasicPassword"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="signup__form__container__group mb-3"
          >
            <label
              class="form-label"
              for="formBasicConfirmPassword"
            >
              Confirm Password
            </label>
            <input
              autocomplete="off"
              class="signup__form__container__group__input form-control"
              id="formBasicConfirmPassword"
              name="confirm"
              placeholder="Confirm Password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="signup__form__container__group mb-3 mt-5"
          >
            <label
              class="form-label"
              for="formBasicRadio"
            >
              Signup as:
            </label>
            <br />
            <div
              class="form-check-inline"
            >
              <input
                checked=""
                class="form-check-input"
                id="customer-radio"
                name="signup-customer"
                type="radio"
                value="Customer"
              />
            </div>
            <label
              class="me-3 form-label"
              for="customer-radio"
            >
              Customer
            </label>
            <div
              class="form-check-inline"
            >
              <input
                class="form-check-input"
                id="admin-radio"
                name="signup-admin"
                type="radio"
                value="Admin"
              />
            </div>
            <label
              class="form-label"
              for="admin-radio"
            >
              Admin
            </label>
          </div>
          <div
            class="mb-5"
          />
          <button
            class="btn btn-primary"
            style="width: 100%; height: 3rem; font-size: 16px; text-transform: uppercase;"
            type="submit"
          >
            Sign up
          </button>
          <div
            class="mt-4"
            style="text-align: center;"
          >
            Already registered? 
            <a
              href="/login"
            >
              Login
            </a>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
